### ISSUE TITLE
basic auth

### ISSUE THREAD

**Issue by burningTyger** [fraserxu/electron-pdf#135]
Wow, electron-pdf has evolved greatly. I'm using basic auth on my server and would like to have the option to pass a username and password to electron-pdf. I could do http://user:pass@example.com but a paramter looks cleaner.

**Comment by codecounselor** [comment]
Thanks @burningTyger, what did you have in mind for a parameter?  When I added support for cookies I took advantage of the [BrowserWindow API](http://electron.atom.io/docs/api/browser-window/).  Since the url has to be set in the browser I'm not sure anything can be done with headers, and I don't know what Electron supports for basic auth.  

Maybe you can see what you think works best and submit a PR?

**Comment by burningTyger** [comment]
Hi, @codecounselor, sorry I'm a ruby guy, can't really help with JS. Right now Im stripping the `http://` part of the URL, add the username and password and prefix the `http://` again to get through basic auth. Not very cool, though.

**Comment by steric85** [comment]
Is there any way to pass a username and password to electron-pdf ?

**Comment by fraserxu** [comment]
what @burningTyger said to pass it in the url ^

**Comment by steric85** [comment]
I did that but it's not working.
Can it be because the password or username already has `@` or `:` ?
@fraserxu

**Comment by fraserxu** [comment]
Maybe you can try to escape the characters if so?

**Comment by steric85** [comment]
Tried just now, it ain't working

**Comment by fraserxu** [comment]
🤷‍♂️

**Comment by fraserxu** [comment]
Can you try the url in browser directly and see if it works?

**Comment by steric85** [comment]
No, the url is not working in the browser with and without escaping the characters.

**Comment by fraserxu** [comment]
Then you should probably try to find out how to make it work in browser first and it should work in electron-pdf.

**Comment by steric85** [comment]
Ow Ohkay. Thanks a lot for giving time to my query! 😄
